pkg/karmadactl/interpret/interpret.go

```diff
@@ -0,0 +1,166 @@
+    
+        # Execute the retention rule for local resource
+        %[1]s interpret -f customization.yml --operation retention --desired-file desired.json --observed-file observed.json
+        # Execute the retention rule for server resource
+        %[1]s interpret -f customization.yml --operation retention --desired-file desired.json --observed deployments/foo`)
+)
+
+const (
+	customizationResourceName = "resourceinterpretercustomizations"
+)
+
+// NewCmdInterpret new interpret command.
+func NewCmdInterpret(f util.Factory, parentCommand string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &Options{
+		IOStreams: streams,
+		Rules:     allRules,
+	}
+	cmd := &cobra.Command{
+		Use:                   "interpret (-f FILENAME | NAME) (--operation OPERATION) [-n NAMESPACE] [--ARGS VALUE]... ",
+		Short:                 "Check the valid of customizations. Or show the result of the rules running",
+		Long:                  interpretLong,
+		SilenceUsage:          true,
+		DisableFlagsInUseLine: true,
+		Example:               fmt.Sprintf(interpretExample, parentCommand),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Annotations: map[string]string{
+			util.TagCommandGroup: util.GroupClusterTroubleshootingAndDebugging,
+		},
+	}
+
+	flags := cmd.Flags()
+	options.AddKubeConfigFlags(flags)
+	flags.StringVarP(options.DefaultConfigFlags.Namespace, "namespace", "n", *options.DefaultConfigFlags.Namespace, "If present, the namespace scope for this CLI request")
+	flags.StringVar(&o.Operation, "operation", o.Operation, "The interpret operation to use. One of: ("+strings.Join(o.Rules.Names(), ",")+")")
+	flags.BoolVar(&o.Check, "check", o.Check, "Check the given customizations")
+	flags.StringVar(&o.Desired, "desired", o.Desired, "The type and name of resource on server to use as desiredObj argument in rule script. For example: deployment/foo.")
+	flags.StringVar(&o.DesiredFile, "desired-file", o.DesiredFile, "Filename, directory, or URL to files identifying the resource to use as desiredObj argument in rule script.")
+	flags.StringVar(&o.Observed, "observed", o.Observed, "The type and name of resource on server to use as observedObj argument in rule script. For example: deployment/foo.")
+	flags.StringVar(&o.ObservedFile, "observed-file", o.ObservedFile, "Filename, directory, or URL to files identifying the resource to use as observedObj argument in rule script.")
+	flags.StringSliceVar(&o.StatusFile, "status-file", o.StatusFile, "Filename, directory, or URL to files identifying the resource to use as statusItems argument in rule script.")
+	flags.Int32Var(&o.DesiredReplica, "desired-replica", o.DesiredReplica, "The desiredReplica argument in rule script.")
+	cmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, "containing the customizations")
+	_ = flags.MarkHidden("kustomize")
```

Comment:
`    -f, --filename=[]:
	Filename, directory, or URL to files containing the customizations`

I see this flag accepts a slice of filenames, is it an expected usage?

By the way, can we get rid of `cmdutil.AddFilenameOptionFlags` and turn to define flags by ourselves? 
Seems we don't need `Kustomize` and `Recursive` that are inherited from that.

Reply:
> I see this flag accepts a slice of filenames, is it an expected usage?

Yes. Since rules can be defined in multiple customizations. Users can interpret from multiple files.
```
karmadactl interpret -f a.yml -f b.yml
```

Reply:
Get it. That makes sense.

Reply:
> Seems we don't need `Kustomize` and `Recursive` that are inherited from that.

`Recursive` option is accepted. Since we accept multiple customizations files.